microprofile/connectors/kafka/src/main/java/io/helidon/microprofile/connectors/kafka/BasicKafkaProducer.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.microprofile.connectors.kafka;
+
+import java.io.Closeable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.logging.Logger;
+
+import io.helidon.config.Config;
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.header.Header;
+
+/**
+ * Basic Kafka producer covering basic use-cases.
+ * Configurable by Helidon {@link io.helidon.config.Config Config},
+ * For more info about configuration see {@link KafkaConfigProperties}.
+ * <p>
+ * Usage:
+ * <pre>{@code new SimpleKafkaProducer<Long, String>("job-done-producer", Config.create())
+ *             .produce("Hello world!");
+ * }</pre>
+ *
+ * @param <K> Key type
+ * @param <V> Value type
+ * @see KafkaConfigProperties
+ * @see io.helidon.config.Config
+ */
+class BasicKafkaProducer<K, V> implements Closeable {
+
+    private static final Logger LOGGER = Logger.getLogger(BasicKafkaProducer.class.getName());
+    private final KafkaConfigProperties properties;
+    private final KafkaProducer<K, V> producer;
+
+    /**
+     * Kafka producer created from {@link io.helidon.config.Config config} under kafka-producerId,
+     * see configuration {@link KafkaConfigProperties example}.
+     *
+     * @param config Helidon {@link io.helidon.config.Config config}
+     */
+    BasicKafkaProducer(Config config) {
+        properties = new KafkaConfigProperties(config);
+        producer = new KafkaProducer<>(properties);
+    }
+
+    /**
+     * Send record to all provided topics,
+     * blocking until all records are acknowledged by broker.
+     *
+     * @param value Will be serialized by <b>value.serializer</b> class
+     *              defined in {@link KafkaConfigProperties configuration}
+     * @return Server acknowledged metadata about sent topics
+     */
+    public List<RecordMetadata> produce(V value) {
+        List<Future<RecordMetadata>> futureRecords =
+                produceAsync(null, null, null, null, value, null);
```

Comment:
Using `null` is not encouraged in Helidon. Would be better to send correct defaults rather than nulls - this is very error prone

Reply:
Done